File: csharp/armstrong-numbers/ArmstrongNumbers.cs
using System;
using System.Linq;
using System.Collections.Generic;

public static class ArmstrongNumbers
{
    public static bool IsArmstrongNumber(int number)
    {
        var str = number.ToString();
        return number == Enumerable.Sum(str.ToCharArray(),
                                        c => Math.Pow(c - '0', str.Length));
    }
}

File: java/armstrong-numbers/src/main/java/ArmstrongNumbers.java
class ArmstrongNumbers {
    boolean isArmstrongNumber(int numberToCheck) {
        var str = Integer.toString(numberToCheck);
        var size = str.length();
        var sum = 0;

        for (char c : str.toCharArray())
            sum += Math.pow(c - '0', size);

        return sum == numberToCheck;
    }
}
